samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,199 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+// https://raw.githubusercontent.com/spmallick/learnopencv/master/Mask-RCNN/mscoco_labels.names
+// It is based on the OpenCV project. It is subject to the license terms in the LICENSE file found in this distribution and at http://opencv.org/license.html
+//
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --video=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ input i     | 0 | Path to input image file or video or camera id.  }"
+    "{ classes n   |object_detection_classes_coco.txt| Path to a text file with names of classes.  }"
+    "{ model w     |mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb | The pre-trained weights.  }"
+    "{ height H    |800| Preprocess input image by resizing to a specific height }"
+    "{ width W     |800| Preprocess input image by resizing to a specific columns }"
+    "{ config c    |mask_rcnn_inception_v2_coco_2018_01_28.pbtxt |The text graph file that has been tuned by the OpenCV’s DNN support group  }"
+    "{ cthr        | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+void static drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask, float maskThreshold, vector<Scalar>& colors, vector<string>& classes);
+
+void static postprocess(Mat& frame, const vector<Mat>& outs, float confThreshold, float maskThreshold, vector<Scalar>& colors, vector<string>& classes);
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("This sample demonstrates instance segmentation network called Mask-RCNN.");
+    if (argc == 1 || parser.has("help"))
```

Comment:
`if (parser.has("help"))` enough here because default values for params are all defined